### ISSUE TITLE
Gatsby-plugin-sharp generates source sets bigger than maxWidth parameter.

### ISSUE THREAD

**Issue by bigfanjs** [gatsbyjs/gatsby#14752]
## Summary
`Gatsby-plugin-sharp` seems like it generates source sets(`srcset`) bigger than what specified in `maxWidth` parameter.

that is mentioned in `gatsby-transformer-sharp` docs:
> If the max width of the container for the rendered markdown file is 800px, the fluid sizes would then be: 200, 400, 800, 1200, 1600, 2400 – enough to provide close to the optimal image size for every device size / screen resolution.

Ref: https://image-processing.gatsbyjs.org/

## Relevant information
my query:
```
{
  contentfulArticle {
    id
    title
    image {
      fluid(maxWidth: 2080) {
        srcSet
      }
    }
  }
}
```

which generates a `srcset` like:
```
//images.ctfassets.net/image-url.jpg?w=520&h=347&q=50 520w,
//images.ctfassets.net/image-url.jpg?w=1040&h=693&q=50 1040w,
//images.ctfassets.net/image-url.jpg?w=2080&h=1387&q=50 2080w,
//images.ctfassets.net/image-url.jpg?w=3120&h=2080&q=50 3120w,
//images.ctfassets.net/image-url.jpg?w=4160&h=2773&q=50 4160w,     <--- this wont work
//images.ctfassets.net/image-url.jpg?w=5184&h=3456&q=50 5184w     <--- this wont work
```

Those images are coming from Contentful, and Contentful does not allow fetching images above `4000` in size. And the issue is that in large screens, the images would fail to load. because the browser would select the last source (`//images.ctfassets.net/image-url.jpg?w=5184&h=3456&q=50 5184w`) in large screens which would fail to load because the width specified as `5184`.

On Contentful docs:
>Specify width & height
You can resize the image to the desired width and height. The maximum allowed value is 4000 pixels. The default is the original image width and height.

Is there a work around to this issue, like preventing the plugin from generating image sources that go above a certain width. or deciding the image size for each breakpoint?

**Comment by KyleAMathews** [comment]
This would be something to fix in gatsby-source-contentful as since they have their own image api, gatsby-plugin-sharp doesn't process images from Contentful. Would you like to add a PR restricting width to 4000px here https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-source-contentful/src/extend-node-type.js ?

**Comment by bigfanjs** [comment]
@KyleAMathews Thank you for the quick answer. Yes I really would love to 😊

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by sidharthachatterjee** [comment]
Published in `gatsby-source-contentful@2.0.72`
